File: repeatedly-until/app/src/androidTest/java/com/sqisland/espresso/repeatedly_until/MainActivityTest.java
package com.sqisland.espresso.repeatedly_until;

import android.support.test.rule.ActivityTestRule;

import org.junit.Rule;
import org.junit.Test;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static android.support.test.espresso.action.ViewActions.repeatedlyUntil;

public class MainActivityTest {
  @Rule
  public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

  @Test
  public void eleven() {
    onView(withId(R.id.number))
        .check(matches(withText("0")));

    onView(withId(R.id.number))
        .perform(repeatedlyUntil(click(), withText("11"), 20));

    onView(withId(R.id.squared))
        .check(matches(withText("121")));
  }
}

File: repeatedly-until/app/src/androidTest/java/com/sqisland/espresso/repeatedly_until/MainActivityTest.kt
package com.sqisland.espresso.repeatedly_until

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.action.ViewActions.repeatedlyUntil
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.rule.ActivityTestRule
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MainActivityTest {
  @get:Rule
  val activityRule: ActivityTestRule<MainActivity> = ActivityTestRule(MainActivity::class.java)

  @Test
  fun eleven() {
    onView(withId(R.id.number))
      .check(matches(withText("0")))

    onView(withId(R.id.number))
      .perform(repeatedlyUntil(click(), withText("11"), 20))

    onView(withId(R.id.squared))
      .check(matches(withText("121")))
  }
}